From f507b3a93f2d76782cb226491163446d2f76c5b6 Mon Sep 17 00:00:00 2001
From: luca abeni <luca.abeni@santannapisa.it>
Date: Thu, 18 Nov 2021 13:18:28 +0000
Subject: [PATCH 08/10] fix find_lock_later_rq

The overload check must be on the selected rq, not on the current one!
---
 kernel/sched/deadline.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/deadline.c b/kernel/sched/deadline.c
index 3dcd06aaf..f55526e87 100644
--- a/kernel/sched/deadline.c
+++ b/kernel/sched/deadline.c
@@ -2153,7 +2153,7 @@ static struct rq *find_lock_later_rq_ff(struct task_struct *task, struct rq *rq)
 		 * its earliest one has a later deadline than our
 		 * task, the rq is a good one.
 		 */
-		if (rq->dl.this_bw + task->dl.dl_bw < rq->dl.max_bw)
+		if (later_rq->dl.this_bw + task->dl.dl_bw < later_rq->dl.max_bw)
 			break;
 
 		/* Otherwise we try again. */
-- 
2.25.1

